src/wovnio/wovnphp/core/tests/unit/WovnURLTest.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+
+namespace Wovnio\Wovnphp\Core\Tests\Unit;
+
+require_once 'src/wovnio/wovnphp/core/exceptions/WovnException.php';
+require_once 'src/wovnio/wovnphp/core/exceptions/WovnLangException.php';
+require_once 'src/wovnio/wovnphp/core/WovnOption.php';
+require_once 'src/wovnio/wovnphp/core/WovnLangDirectory.php';
+require_once 'src/wovnio/wovnphp/core/WovnLang.php';
+require_once 'src/wovnio/wovnphp/core/url_handlers/WovnURLHandler.php';
+require_once 'src/wovnio/wovnphp/core/url_handlers/WovnQueryURLHandler.php';
+require_once 'src/wovnio/wovnphp/core/url_handlers/WovnPathURLHandler.php';
+require_once 'src/wovnio/wovnphp/core/url_handlers/WovnSubdomainURLHandler.php';
+require_once 'src/wovnio/wovnphp/core/WovnURL.php';
+
+use Wovnio\Wovnphp\Core\WovnLangDirectory;
+use Wovnio\Wovnphp\Core\WovnOption;
+use Wovnio\Wovnphp\Core\WovnURL;
+
+class WovnURLTest extends \PHPUnit_Framework_TestCase
```

Comment:
I don't see a purpose for WovnURL at the moment, versus just having WovnUrlHandler with potentially an abstract base class for common logic. At the moment it isn't actually doing anything except delegate to WovnURL.

Ideally, as far as tests go we should be testing the url handlers individually. Having all patterns in a single file makes it so messy.